kosat-core/src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -727,6 +760,17 @@ class CDCL {
         db.add(clause)
         watchers[clause[0]].add(clause)
         watchers[clause[1]].add(clause)
+        if (clause.learnt) dratBuilder.addClause(clause)
+    }
+
+    /**
+     * Remove [clause] from the database. Watchers will be detached
+     * later, during [ClauseDatabase.reduceIfNeeded].
+     */
+    fun detachClause(clause: Clause) {
```

Comment:
This method does not "detach the clause from the database", it only marks it as garbage. This is a separate question whether such marking should be considered an actual deletion for DRAT, but probably yes. Consider naming this method as "deleteClause" or "markDeleted"